p2p/p2p.go

```diff
@@ -45,17 +44,10 @@ type Service struct {
 	runLock sync.RWMutex
 }
 
-func New(
-	addr,
-	userAgent,
-	bootPeers,
-	privKeyStr string,
-	snNetwork utils.Network,
-	log utils.SimpleLogger,
-) (*Service, error) {
+func New(addr, userAgent, bootPeers, privKeyStr string, snNetwork utils.Network, log utils.SimpleLogger) (*Service, error) {
 	if addr == "" {
-		// 0.0.0.0 will listen on any interface device.
-		addr = fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", defaultSourcePort)
+		// 0.0.0.0/tcp/0 will listen on any interface device and assing a free port.
```

Comment:
```suggestion
		// 0.0.0.0/tcp/0 will listen on any interface device and assign a free port.
```

Not a problem with this PR, but I am concerned about making the p2p server accessible over the network by default. What do you think about defaulting to `127.0.0.1`? I'm not convinced that's a reasonable default either, though.

Reply:
It is fine for now.

Reply:
Why are we listening on a random port?